Route based on distance

monteirodom opened the pull request:
Being able to re-order way-points to optimize route

Reordering waypoints and compute route based on shortest distance.
- Reordering waypoints : add `optimize=true` to the list of waypoints
`waypoints=optimize=true|43.2345,2,3445|42.5564,3,76234`
- Use shortest distance : add an optional parameter path=shortest

Related PRs
- Distance: https://github.com/WebGeoServices/distance/pull/110
- PyLibOSRM: https://github.com/WebGeoServices/pylibosrm/pull/12

monteirodom committed:
Fix : duration values were of type EdgeWeight => make them of type EdgeDuration

monteirodom committed:
adapt lua version

monteirodom committed:
Extension to Trip ... DirectShortPathSearch with lambda function to compute weights (computed based on option)

monteirodom committed:
AlternativePathSearch with lambda function to compute weights (computed based on option)

monteirodom committed:
AlternativePathSearch with lambda function to compute weights (computed based on option)

monteirodom committed:
Continue ... AlternativePathSearch with lambda function to compute weights (computed based on option)

monteirodom committed:
Move routing option to base parameters, make it more general than just for route

monteirodom committed:
Adaptation to be able to search for routes based on a dynamic optimization parameter (optimize=default | distance | time)
Based on usage of lambda function passed around to compute weights

monteirodom committed:
Merge branch 'fix_type_duration'

monteirodom committed:
Revert "AlternativePathSearch with lambda function to compute weights (computed based on option)"

This reverts commit 194450f4edb8fe756cc137b0a6cb10d1f33c7e01.

monteirodom committed:
Route Based on distance (or another startegy)ev/update-v5.26

monteirodom committed:
Missing '{'

monteirodom committed:
Missing '{'

monteirodom committed:
Update weightName according to request option
Fix for_loop variable ref so that the tests compile

monteirodom committed:
Merge branch 'route_based_on_distance' of /Users/monteirodom/PycharmProjects/osrm-backend into route_based_on_distance

monteirodom committed:
Patching version for new functionalities

monteirodom committed:
Adapt test

monteirodom committed:
Adapt test

sebastienpro commented on include/engine/api/route_api.hpp:
```diff
@@ -61,7 +62,7 @@ class RouteAPI : public BaseAPI
         {
             auto &json_result = response.get<util::json::Object>();
-            MakeResponse(raw_routes, all_start_end_points, json_result);
+            MakeResponse(raw_routes, all_start_end_points, weightName, json_result);
```
Toutes les variables sont en snake_case ça serait mieux de garder ce formalisme

sebastienpro commented on include/engine/routing_algorithms.hpp:
```diff
@@ -19,22 +19,32 @@ namespace engine
 class RoutingAlgorithmsInterface
 {
   public:
+    //virtual std::function<EdgeWeight(const EdgeID id, const EdgeID turnId)>
```
Mieux vaut éviter de garder du code commenté, si il n'est pas utile autant le virer

monteirodom committed:
Missing function and signatures to passe engine-tests

monteirodom committed:
Missing function and signatures to passe engine-tests

monteirodom committed:
Review fixes :
- all variables in snake_case
- deleting commented code

monteirodom committed:
Review fixes :
- all variables in snake_case
- deleting commented code

monteirodom committed:
Versioning, changelog and API (http) docs

monteirodom committed:
Versioning, changelog and API (http) docs

monteirodom committed:
Merge commit 'f16e0e0a1efff8f09d0faa86a56da5eb2ae69e64' into route_based_on_distance

monteirodom committed:
Versioning

monteirodom committed:
Fix : node_weight function missing in search for match service

monteirodom committed:
Internal WGS versioning, to be forward compatible with initial versioning : use -GWSn for internal developpement

monteirodom committed:
Being able to run the tests from the container
Fix : essentially, the test folder was on .dockerignore, thus preventing all the tests to run
Integrated a tester container within the Dockerfile, to run the tests

monteirodom committed:
Being able to run the tests from the container
Fix : essentially, the test folder was on .dockerignore, thus preventing all the tests to run
Integrated a tester container within the Dockerfile, to run the tests

sebastienpro commented on Dockerfile:
```diff
@@ -16,14 +16,37 @@ WORKDIR /usr/src/app
+FROM builder as tester
+
+ARG DEBIAN_FRONTEND="noninteractive"
+ARG VERSION="0.0.0"
```
Version 0.0.0 je ne suis pas sur ;)

Commit:
Normalise the testing
Use ./run_tests.sh to docker-compose test
Add this test the the github actions

Commit:
Normalise the testing
Use ./run_tests.sh to docker-compose test
Add this test the the github actions

monteirodom committed:
Adapt function signature to test facade

monteirodom committed:
Fix tests to be up to date
- Timezone checks moved on commit #564a29141
- Timezone checks use lower_case
- Profile path was missing on testing a valid config
- Check of polygon coordinates array was wrong : no check was made for 3-dimensional array

monteirodom committed:
Integrate boost tests on the test suite

monteirodom committed:
Updating internal versioning

monteirodom committed:
Merge branch 'route_based_on_distance' of github.com:WebGeoServices/osrm-backend into route_based_on_distance

# Conflicts:
#	test.sh

monteirodom committed:
Updating versioning field.

Commit:
Add Cache to the Dockerfile for the build folder

Commit:
Fix build
Use docker-compose instead of docker compose

Commit:
Fix build
Add build context

monteirodom committed:
Option to skip the roundtrip path on the response to a trip request.
Following review https://github.com/WebGeoServices/pylibosrm/pull/12#discussion_r704327837

monteirodom committed:
Strip the roundtrip part of a trip, according to the option parameter

monteirodom committed:
Make base_parameters visible to wrapper

monteirodom committed:
Tests for parameters : optimize and skip_roundtrip
- optimize : which optimization (distance, time, weight)
- skip_roundtrip : remove round trip part from the response even when roundtrip option is mandatory (used when optimizing the waypoints, but no roundtrip is actually desired).
